libraries/helpers.rb

```diff
@@ -1,101 +1,100 @@
-def service_user
-  value_for_platform_family(
-    %w(suse fedora rhel amazon) => 'memcached',
-    'debian' => 'memcache',
-    'default' => 'nobody'
-  )
-end
+module Memcached
+  module Helpers
+    def service_user
+      value_for_platform_family(
+        %w(suse fedora rhel amazon) => 'memcached',
+        'debian' => 'memcache',
+        'default' => 'nobody'
+      )
+    end
 
-def service_group
-  value_for_platform_family(
-    %w(suse fedora rhel amazon) => 'memcached',
-    'debian' => 'memcache',
-    'default' => 'nogroup'
-  )
-end
+    def service_group
+      value_for_platform_family(
+        %w(suse fedora rhel amazon) => 'memcached',
+        'debian' => 'memcache',
+        'default' => 'nogroup'
+      )
+    end
 
-def binary_path
-  new_resource.binary_path || value_for_platform_family(
-    'suse' => '/usr/sbin/memcached',
-    'default' => '/usr/bin/memcached'
-  )
-end
+    def binary_path(new_resource)
+      new_resource.binary_path || value_for_platform_family(
+        'suse' => '/usr/sbin/memcached',
+        'default' => '/usr/bin/memcached'
+      )
+    end
 
-def lock_dir
-  value_for_platform_family(
-    %w(rhel fedora suse amazon) => '/var/lock/subsys',
-    'default' => '/var/lock'
-  )
-end
+    def lock_dir
+      value_for_platform_family(
+        %w(rhel fedora suse amazon) => '/var/lock/subsys',
+        'default' => '/var/lock'
+      )
+    end
 
-# if the instance name is memcached don't spit out memcached_memcached
-def memcached_instance_name
-  new_resource.instance_name == 'memcached' ? 'memcached' : "memcached_#{new_resource.instance_name}"
-end
+    # if the instance name is memcached don't spit out memcached_memcached
+    def memcached_instance_name(new_resource)
+      new_resource.instance_name == 'memcached' ? 'memcached' : "memcached_#{new_resource.instance_name}"
+    end
 
-def disable_default_memcached_instance
-  return unless new_resource.disable_default_instance
+    def disable_default_memcached_instance(new_resource)
+      return unless new_resource.disable_default_instance
 
-  service 'disable default memcached' do
-    service_name 'memcached'
-    action [:stop, :disable]
-    not_if { new_resource.instance_name == 'memcached' }
-  end
-end
+      service 'disable default memcached' do
+        service_name 'memcached'
+        action [:stop, :disable]
+        not_if { new_resource.instance_name == 'memcached' }
+      end
+    end
 
-def remove_default_memcached_configs
-  return unless new_resource.remove_default_config
+    def remove_default_memcached_configs(new_resource)
```

Comment:
If you include these helpers into the action_class they should get the new_resource automatically